tests/utf8.cpp

```diff
@@ -12,6 +12,13 @@
 
 // This code tests that Unicode std::wstrings can be marshalled between C++ and JS.
 int main() {
+  const char latin1String[] = "\x26\xA0\xF7";
+  EM_ASM({
+    var str = Module.AsciiToString($0);
+    var expected = "\\x26\\xA0\\xF7";
+    if (str !== expected) throw 'string mismatch: "' + str + '" != "' + expected + '"';
```

Comment:
can call `assert(str == expected, 'string mismatch [..]`

Reply:
Is `assert` available inside the JavaScript? Note that this wouldn't fail outside of the JS block, because from C/C++ point of view, both are decoded to the same array of `char`s after round-trippping, while in JS they are different strings.

I suspect this is the reason this slipped by in the past.

Reply:
We define assert in JS, yes (the optimizer might remove it if it's not used, though).

Reply:
Ah ok then. FWIW I was just looking at other assertions in the same file, which also use custom comparisons.

Reply:
Updated.